packages/github-tinacms-content/src/sourceProviderConnection.ts

```diff
@@ -0,0 +1,34 @@
+export interface SourceProviderConnection {
+  forkFullName: string
+  headBranch: string
+}
+
+export interface Response {
+  accessToken: string | null
+  sourceProviderConnection: SourceProviderConnection | null
+}
+
+export interface PreviewData {
+  fork_full_name: string
+  head_branch: string
+  github_access_token: string
+}
+
+export const getGithubDataFromPreviewProps = (
+  previewData?: PreviewData
+): Response => {
+  if (!previewData) {
+    return {
+      sourceProviderConnection: null,
+      accessToken: null,
+    }
+  }
+
+  return {
+    accessToken: previewData.github_access_token,
+    sourceProviderConnection: {
+      forkFullName: previewData.fork_full_name,
+      headBranch: previewData.head_branch || 'master',
+    },
+  }
+}
```

Comment:
This feels really Next.js specific. Maybe belongs in `next-tinacms-github`, or maybe it needs to be renamed?

Reply:
true, or maybe even just within the site for now.

Reply:
Moved it back into tinacms.org